projects/scripts/project-lattice.mk

```diff
@@ -0,0 +1,162 @@
+M_DEPS += $(wildcard *system_constr.sdc)
+
+PB_DEPS_FILTER += %.tcl
+PB_DEPS_FILTER += %.mem
+PB_DEPS_FILTER_OUT += %system_project.tcl
+PB_DEPS_FILTER_OUT += %adi_project_lattice.tcl
+
+R_DEPS_FILTER += %.v
+R_DEPS_FILTER += %.vhdl
+R_DEPS_FILTER += %.sdc
+R_DEPS_FILTER += %.pdc
+R_DEPS_FILTER += %.mem
+R_DEPS_FILTER += %.ipx
+R_DEPS_FILTER += %adi_env.tcl
+R_DEPS_FILTER += %adi_project_lattice.tcl
+R_DEPS_FILTER += %system_project.tcl
+
+PB_TARGETS += $(PROJECT_NAME)/$(PROJECT_NAME)/$(PROJECT_NAME).sbx
+PB_TARGETS += $(PROJECT_NAME)/$(PROJECT_NAME)/$(PROJECT_NAME).v
+
+R_TARGETS += $(PROJECT_NAME)/$(PROJECT_NAME).rdf
+R_TARGETS += $(PROJECT_NAME)/impl_1/$(PROJECT_NAME)_impl_1.bit
+
+.PHONY: all pb rd rd-force pb-force clean clean-all clean-pb clean-rd
+# watch-pb watch-rd
+
+all: pb rd
+
+pb: $(PB_TARGETS)
+
+rd: $(R_TARGETS)
+
+clean:
+	-rm -Rf ${PROJECT_NAME}
+	-rm -f $(wildcard *.log)
+	-rm -Rf ./ipcfg
+	-rm -Rf ./sge
+
+clean-all:
+	-rm -Rf ${PROJECT_NAME}
+	-rm -f $(wildcard *.log)
+	-rm -Rf ./ipcfg
+	-rm -Rf $(filter-out . .. ./. ./.., $(wildcard .*))
+	-rm -Rf ./sge
+
+clean-pb:
+	-rm -Rf $(wildcard $(PROJECT_NAME)/$(PROJECT_NAME)/*)
+	-rm -f $(PROJECT_NAME)/.socproject
+	-rm -Rf ./ipcfg
+	-rm -f $(PROJECT_NAME)_propel_builder.log
+	-rm -Rf ./sge
+
+clean-rd:
+	-rm -Rf $(filter-out $(PROJECT_NAME)/$(PROJECT_NAME) \
+		$(PROJECT_NAME)/.socproject, \
+		$(wildcard $(PROJECT_NAME)/*))
+	-rm -Rf $(filter-out $(PROJECT_NAME)/$(PROJECT_NAME) \
+		$(PROJECT_NAME)/.socproject \
+		$(PROJECT_NAME)/. \
+		$(PROJECT_NAME)/.., $(wildcard $(PROJECT_NAME)/.*))
+	-rm -f $(PROJECT_NAME)_radiant.log
+
+$(PB_TARGETS): $(filter-out $(PB_DEPS_FILTER_OUT),$(filter $(PB_DEPS_FILTER), $(M_DEPS)))
+	-rm -f $(PROJECT_NAME)_propel_builder.log
+	$(call build, \
+		$(PROPEL_BUILDER) system_project_bd.tcl ${PROJECT_NAME}, \
+		$(PROJECT_NAME)_propel_builder.log, \
+		$(HL)$(PROJECT_NAME)$(NC) project)
+
+$(R_TARGETS): $(filter $(R_DEPS_FILTER), $(M_DEPS))
+	@for file in $(filter $(R_DEPS_FILTER), $(M_DEPS)); do \
+		if [ ! -f $$file ]; then \
+			echo "No [$(HL)$$file$(NC)] found. ... $(RED)FAILED$(NC)"; \
+			echo "For details see $(HL)$(CURDIR)/$(PROJECT_NAME)_propel_builder.log$(NC)"; \
+			exit 2; \
+		fi \
+	done
+	-rm -f $(PROJECT_NAME)_radiant.log
+	$(call build, \
+		$(RADIANT) system_project.tcl ${PROJECT_NAME}, \
+		$(PROJECT_NAME)_radiant.log, \
+		$(HL)$(PROJECT_NAME)$(NC) project)
+
+pb-force:
+	$(call build, \
+	$(PROPEL_BUILDER) system_project_bd.tcl ${PROJECT_NAME}, \
+	$(PROJECT_NAME)_propel_builder.log, \
+	$(HL)$(PROJECT_NAME)$(NC) project)
+
+rd-force:
+	$(call build, \
+	$(RADIANT) system_project.tcl ${PROJECT_NAME}, \
+	$(PROJECT_NAME)_radiant.log, \
+	$(HL)$(PROJECT_NAME)$(NC) project)
+
+# To easily check included project dependencies for Propel Builder and Radiant.
+# watch-pb:
+# 	echo $(filter-out $(PB_DEPS_FILTER_OUT),$(filter $(PB_DEPS_FILTER), $(M_DEPS)))
+# watch-rd:
+# 	echo $(filter $(R_DEPS_FILTER), $(M_DEPS))
```

Comment:
We should not commit commented code, decide if we include or not

Reply:
It is fixed from 'Update version 4'. [project-lattice.mk](https://github.com/analogdevicesinc/hdl/blob/lattice_build_bd/projects/scripts/project-lattice.mk#L157)